# ackermann_controller_test.yaml
my_car:
  # Publish all joint states --公布所有--------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50    #左驱动轮
  
  # Velocity Controllers ----速度控制器---------------------
  back_left_wheel_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: back_left_wheel2base_link
    pid: {p: 100.0, i: 0.1, d: 0.0, i_clamp: 0.0}
  back_right_wheel_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: back_right_wheel2base_link
    pid: {p: 100.0,i: 0.1, d: 0.0, i_clamp: 0.0}
  front_left_wheel_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: front_left_wheel2front_left_wheel_joint
    pid: {p: 80.0, i: 0.1, d: 0.0, i_clamp: 0.0}
  front_right_wheel_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: front_right_wheel2front_right_wheel_joint
    pid: {p: 80.0, i: 0.1, d: 0.0, i_clamp: 0.0}

  # Position Controllers ---位置控制器-----------------------
  front_left_wheel_joint_position_controller:
    joint: front_left_wheel_joint2base_link
    type: effort_controllers/JointPositionController
    pid: {p: 70.0, i: 0.1, d: 0.5}
  front_right_wheel_joint_position_controller:
    joint: front_right_wheel_joint2base_link
    type: effort_controllers/JointPositionController
    pid: {p: 70.0, i: 0.1, d: 0.5}
    